src/main/java/ch/njol/skript/config/SectionNode.java

```diff
@@ -130,66 +138,78 @@ public Node remove(final String key) {
 		n.parent = null;
 		return n;
 	}
-	
+
 	/**
-	 * Iterator over all non-void nodes of this section.
+	 * Gets the node at the specified index. May be null.
+	 * The index includes all nodes, including void nodes.
+	 *
+	 * @param idx The index of the node to get
+	 * @return The node at the specified index. May be null.
+	 * @throws IllegalArgumentException if the index is out of bounds
+	 */
+	@Nullable Node getAt(int idx) {
+		Preconditions.checkArgument(idx >= 0 && idx < size(), "idx out of bounds: %s", idx);
+		return nodes.get(idx);
+	}
+
+	/**
+	 * @return An iterator over all non-void nodes in this section.
 	 */
 	@Override
-	public Iterator<Node> iterator() {
-		@SuppressWarnings("null")
-		@NotNull
-		final Iterator<Node> iter = nodes.iterator();
-		return new CheckedIterator<Node>(iter, new NullableChecker<Node>() {
-			@Override
-			public boolean check(final @Nullable Node n) {
-				return n != null && !n.isVoid();
-			}
-		}) {
+	public @NotNull Iterator<Node> iterator() {
+		return new CheckedIterator<>(fullIterator(),
+			n -> Objects.nonNull(n) && !n.isVoid()); // double null check to avoid warning
```

Comment:
I think it would be fine to suppress the warning instead with a comment explaining why